### PR TITLE
activities cleanup

### PR DESCRIPTION
**Pull request by Jibz1**
- Remove unused activity code
- Add return to the previous route from the receipt page
-  Use both Transfer amounts from `Pay, PaySui and TransferSui` and EventSummery to get the transfer amount

https://user-images.githubusercontent.com/8676844/216680350-0a00bb4e-6924-4652-903b-546038779b69.mov

### CODE REVIEW

File: apps/wallet/src/ui/app/components/receipt-card/TxnAmount.tsx [hunk 1]
```diff
@@ -11,9 +11,11 @@ type TxnAmountProps = {
     label: string;
 };
 
+// dont show amount if it is 0
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/components/receipt-card/TxnAmount.tsx:14]: I know it's not a common use-case, but is there a reason we want to hide the amount when someone sends sui to themselves? amount still seems useful in activity feed to me


**this PR**
image

**main**
image

File: apps/wallet/src/ui/app/components/receipt-card/index.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ import { TxnAddress } from '_components/receipt-card/TxnAddress';
 import { TxnAmount } from '_components/receipt-card/TxnAmount';
 import { TxnGasSummery } from '_components/receipt-card/TxnGasSummery';
 import { UnStakeTxnCard } from '_components/receipt-card/UnstakeTxnCard';
-import { getTxnEffectsEventID } from '_components/transactions-card/Transaction';
+import { getTxnEffectsEventID } from '_components/transactions-card';
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/components/receipt-card/index.tsx:26]: design looks off here:

**Figma**
image

**PR**
image

File: apps/wallet/src/ui/app/pages/home/receipt/index.tsx [hunk 3]
```diff
@@ -33,13 +31,10 @@ function ReceiptPage() {
         { enabled: !!transactionId, retry: 8 }
     );
 
-    // return route or default to transactions
-    const linkTo = fromRoute ? `/${fromRoute}` : '/transactions';
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/pages/home/receipt/index.tsx:37]: do we use this anywhere?

File: apps/wallet/src/ui/app/components/transactions-card/index.tsx [hunk 4]
```diff
@@ -1,172 +1,216 @@
...
-                            <div className={st.txAmount}>
-                                {formattedAmount} <span>{symbol}</span>
+                <div className="flex flex-col w-full gap-1.5">
+                    {error ? (
+                        <div className="flex flex-col w-full gap-1.5">
+                            <Text color="gray-90" weight="semibold">
+                                Transaction Failed
+                            </Text>
+                            <div className="flex break-all text-issue-dark text-subtitle">
+                                {error}
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/components/transactions-card/index.tsx:167]: can we use `Text` component here?

**Review by mamos-mysten — CHANGES_REQUESTED** [review]
few things I noticed on the first pass

File: apps/wallet/src/ui/app/components/receipt-card/TxnAmount.tsx (hunk 1 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/receipt-card/TxnAmount.tsx:14]: > I know it's not a common use-case, but is there a reason we want to hide the amount when someone sends sui to themselves? amount still seems useful in activity feed to me
> 
> **this PR** image
> 
> **main** image

It is zero

File: apps/wallet/src/ui/app/components/receipt-card/index.tsx (hunk 2 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/receipt-card/index.tsx:26]: Thats when the value is 0

File: apps/wallet/src/ui/app/pages/home/receipt/index.tsx (hunk 3 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/pages/home/receipt/index.tsx:37]: This was removed

File: apps/wallet/src/ui/app/components/transactions-card/index.tsx (hunk 4 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/transactions-card/index.tsx]: done.

File: apps/wallet/src/ui/app/components/receipt-card/index.tsx (hunk 2 above)
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/components/receipt-card/index.tsx:26]: the top margin looks incorrect when we hide the amount: 

image

File: apps/wallet/src/ui/app/pages/home/receipt/index.tsx (hunk 3 above)
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/pages/home/receipt/index.tsx:37]: I mean - is there a need for this anymore? I don't see anything routing with a `from=` param, but just double checking that isn't needed

File: apps/wallet/src/ui/app/components/receipt-card/index.tsx (hunk 2 above)
Comment (Jibz1) [reply apps/wallet/src/ui/app/components/receipt-card/index.tsx:25]: Done, please review

**Review by mystie711 — APPROVED** [review]

**Review by pchrysochoidis — APPROVED** [review]

**Pull request merged**